allennlp/semparse/worlds/atis_world.py

```diff
@@ -46,9 +21,12 @@ class AtisWorld():
         A list of utterances in the interaction, the last element in this list is the
         current utterance that we are interested in.
     """
-    sql_table_context = SqlTableContext(TABLES)
```

Comment:
Why did you remove this as a class variable?  Because you didn't want to have a shared sqlite connection?  I suppose it's possible you somehow got into a race condition in your beaker experiment, depending on how exactly you did the evaluation.

But, either way, I still think the `sql_table_context` should just be constructed inside the `World`, not taken as a dependency.  It's not something that ever changes, so having it as a constructor parameter doesn't add anything.

Reply:
I agree that it is best if `sql_table_context` is constructed inside `World`, but since we constrain strings based on values in the database, the `sql_table_context` now relies on a call to the database, which means we need to pass it the location of the database. It's passed in as a parameter of the dataset reader right now (https://github.com/allenai/allennlp/blob/9a25f994311310e49f695296863f761ca82f0d06/allennlp/data/dataset_readers/atis.py#L78), is there a way to keep it as a class variable and also pass it the location of the database?

Reply:
Ah, yes, that's a good point that I forgot about.  Still, I'd probably just pass the database location in as a parameter, and construct the `SqlTableContext` internally, so it's more obvious what's a configurable dependency and what isn't.

Reply:
I see -- so pass the database location to `AtisWorld` instead of passing a `SqlTableContext`. Would this be inefficient because we're instantiating a `SqlTableContext` for every world?

Reply:
That's a good question, and I don't know what the practical implications are, especially with multiple hits to a sqlite db.  If you think it makes more sense to do it the way it currently is, that's fine.